### ISSUE TITLE
add d3-lasso-plugin

### ISSUE THREAD

**Issue by timelyportfolio** [juba/scatterD3#19]
As an experiment in this [forked branch](https://github.com/timelyportfolio/scatterD3/tree/feature/lasso), I started to add some support for [D3-Lasso-Plugin](https://github.com/skokenes/D3-Lasso-Plugin) to `scatterD3` through an additional argument `lasso`.  Before I added additional features such as `Shiny`/`crosstalk`, I wanted to see how you felt about this new "feature".  Note, the major issue is it interferes with the zoom/drag chart and drag label functionality.  I know `plotly` and `bokeh` add a toolbar for this reason, but I would assume that is out of the scope for this focused library.

Here is a live [example](http://bl.ocks.org/timelyportfolio/c28bff74fbb9c59ec873).

![scatterd3-lasso](https://cloud.githubusercontent.com/assets/837910/12364483/6874f5a0-bb94-11e5-8796-0a4dbd9845e9.gif)

**Comment by juba** [comment]
First, I'm glad you're considering `scatterD3` as a basis for experimentation. So thanks for that :-) 

Adding lasso feature could certainly be useful for inter-widgets communication. It seems perfectly natural and useful to be able to select points on a scatterplot, and I already see potential applications for my `explor` package.

I took a quick look at your code, and it seems to integrate nicely with the existing codebase. Maybe I would prefer highlighting the selected points by changing their opacity rather than their size, but it surely is a detail.

So, you're right, the only drawback is the interaction with zooming and dragging. I'd really prefer zoom/drag to be always available, even if `lasso=TRUE`. Adding a toolbar would certainly help solve this (or rather than a toolbar, a small menu such as the one in `ggvis`). And I could put the "Reset Zoom" and "Export to SVG" controls into it.

So, in summary, I can certainly imagine this lasso feature integrated in `scatterD3` if we find a way to make it compatible with zoom/drag.

Thanks again !

**Comment by timelyportfolio** [comment]
Another option might be a `shift` key modifier, so lasso works when drag + shift as in this imperfect example http://bl.ocks.org/timelyportfolio/4a23f4d57a9a302701cc.  We can change opacity/size however you would like.

**Comment by juba** [comment]
Yes, that's a good idea. Do you think it could be possible to activate it by shift-clicking, and in the same time provide a control which could toggle lasso on/off and could be bound to a button in a shiny interface (in the same way controls are provided for zoom reset and SVG export) ?  Providing a way to add a toggle button could be more intuitive in certain cases that having to explain the shift+clicking mechanism in plain text.

Moreover, this could make sense as I imagine that lasso would be useful essentialy inside shiny apps ?

I think I'd prefer to have selected points put at opacity 1, and others at something like 0.5. Opacity seems more natural to me to highlight points than size (and points size can already be mapped to a variable in `scatterD3`, which could be confusing).

**Comment by juba** [comment]
I just merged the lasso plugin integration, based on your work, in `master`. There are still some things to adjust, and documentation to write, but it's mostly working.

I close this issue now, thanks a lot for your help !
